test/artifact_test.js

```diff
@@ -227,18 +228,79 @@ suite('Artifacts', function() {
       debug('Fetching artifact from: %s', artifactUrl);
       let artifact = await getWithoutRedirecting(artifactUrl);
 
-      let expectedUrl = 
+      let expectedUrl =
         `https://test-bucket-for-any-garbage.s3-us-west-2.amazonaws.com/${taskId}/0/public/singlepart.dat`;
       assume(artifact.headers).has.property('location', expectedUrl);
 
       await verifyDownload(artifact.headers.location, bigfilehash, bigfilesize);
 
     });
 
+    test('S3 single part complete flow, content-encoding: gzip', async () => {
+      const taskId = slugid.v4();
+      const data = crypto.randomBytes(/*12 * 1024 * 1024 +*/ 21);
+      const gzipped = zlib.gzipSync(data);
+
+      debug('### Creating task');
+      await helper.queue.createTask(taskId, taskDef);
+
+      debug('### Claiming task');
+      await helper.queue.claimTask(taskId, 0, {
+        workerGroup:    'my-worker-group',
+        workerId:       'my-worker',
+      });
+
+      debug('### Create artifact');
+      const {
+        requests,
+      } = await helper.queue.createArtifact(taskId, 0, 'public/singlepart.dat', {
+        storageType: 'blob',
+        expires: taskcluster.fromNowJSON('1 day'),
+        contentType: 'binary/octet-stream',
+        contentEncoding: 'gzip',
+        contentLength: data.length,
+        contentSha256: crypto.createHash('sha256').update(data).digest('hex'),
+        transferLength: gzipped.length,
+        transferSha256: crypto.createHash('sha256').update(gzipped).digest('hex'),
+        parts: [{
+          size: gzipped.length,
+          sha256: crypto.createHash('sha256').update(gzipped).digest('hex'),
+        }],
+      });
+
+      debug('### Put first part of artifact');
+      const {method, url, headers} = requests[0];
```

Comment:
Please match the way that all the other blob-type unit tests run uploads instead of reimplementing it.

```javascript
      let uploadOutcome = await client.runUpload(response.requests, uploadInfo);
```